File: server/src/main/java/com/grpc/empuje_comunitario/domain/user/UserId.java
package com.grpc.empuje_comunitario.domain.user;

public final class UserId {
    private final String value;

    private UserId(String value) {
        this.value = value;
    }

    public static UserId newId() {
        return new UserId(java.util.UUID.randomUUID().toString());
    }

    public String value() {
        return value;
    }
}

File: server/src/main/java/com/grpc/empuje_comunitario/domain/user/UserId.kt
package com.grpc.empuje_comunitario.domain.user

import java.util.UUID

data class UserId  constructor(val value: String) {
    companion object {
        fun newId(): UserId {
            return UserId(UUID.randomUUID().toString())
        }

        fun from(value: String): UserId {
            return UserId(value)
        }
    }
}